src/stations.rs

```diff
@@ -26,11 +26,57 @@ impl StationCatalogue {
 
     /// Find the station nearest to the given coordinates.
     pub fn find_near(&self, location: &model::Coordinates) -> &Station {
-        unimplemented!()
+        // use std::iter::Iterator;
+        use crate::compute::gcd::great_circle_distance;
+        let cmp_distance = |s1: &&Station, s2: &&Station| {
+            let d1 = great_circle_distance(&s1.coordinates, location);
+            let d2 = great_circle_distance(&s2.coordinates, location);
+            d1.partial_cmp(&d2).expect("Distances shouldn't be NaN")
+        };
+        self.stations
+            .iter()
+            .min_by(cmp_distance)
+            .expect("StationCatalogue has at least one station, so there must be a minimum")
     }
 
     /// Add a station's data to this catalogue, assigning it an appropriate unique id.
     fn add(&mut self, name: &str, coords: &model::Coordinates) {
-        unimplemented!()
+        let id: u64 = self.stations.len() as u64;
```

Comment:
Do you need both the type ascription and the `as` cast?

Reply:
Good catch; I only need the cast.